src/main/java/io/hyperfoil/tools/qdup/cmd/impl/Sh.java

```diff
@@ -126,7 +127,17 @@ public void postRun(String output,Context context){
                     if (context.checkExitCode() && !isIgnoreExitCode()) {
                         boolean couldBeCtrlC = walk(CmdLocation.createTmp(), (cmd) -> cmd instanceof CtrlC).stream().anyMatch(Boolean::booleanValue);
                         if( !couldBeCtrlC) {
-                            context.error("aborting run due to exit code "+response);
+                            Cmd cmd = this;
+                            StringBuilder stack = new StringBuilder();
+                            while(cmd!=null){
+                                if( !(cmd instanceof ScriptCmd) ){
+                                    stack.append(System.lineSeparator());
+                                    stack.append((cmd instanceof Script ? "script: ":"") + cmd.toString());
```

Comment:
If you know source file and line number it would be useful to add as well.

Reply:
@willr3 I created a debug branch to allow for attaching a remote debugger, I needed to add src line number to cmd: https://github.com/johnaohara/qDup/commit/363d69dcf1d396edb2a26d16d17b748562577760#diff-90dea6208135aed7bb00f049707779cd05a1794ae6ddfc14a109dbfc688024bf